doc/services/llext/build.rst

```diff
@@ -0,0 +1,192 @@
+run at the specified build step and can refer to the properties of the target
+for build-specific details.
+
+The function signature is:
+
+.. code-block:: cmake
+
+   add_llext_command(
+       TARGET <target_name>
+       [PRE_BUILD | POST_BUILD | POST_PKG]
+       COMMAND <command> [args...]
+   )
+
+The different build steps are:
+
+``PRE_BUILD``
+
+    Before the extension code is linked, if the architecture uses dynamic
+    libraries. This step can access `lib_target` and its own properties.
+
+``POST_BUILD``
+
+    After the extension code is built, but before packaging it in an .llext
+    file. This step is expected to create a `pkg_input` file by reading the
+    contents of `lib_output`.
+
+``POST_PKG``
+
+    After the extension output file has been created. The command can operate
+    on the final llext file `pkg_output`.
+
+Anything else after ``COMMAND`` will be passed to ``add_custom_command()`` as-is
+(including multiple commands and other options).
+
+.. _llext_build_edk:
+
+LLEXT Extension Development Kit
+*******************************
+
+When building extensions as a standalone project, outside of the main Zephyr
+build system, it's important to have access to the same set of generated
+headers and compiler flags used by the main Zephyr application, since they have
+a direct impact on how Zephyr headers are interpreted and the extension is
+compiled in general.
+
+This can be achieved by asking Zephyr to generate an Extension Development Kit
+(EDK) from the build artifacts of the main Zephyr application, by running the
+following command which uses the ``llext-edk`` target:
+
+.. code-block:: shell
+
+    west build -t llext-edk
+
+The generated EDK can be found in the build directory under the ``zephyr``
+directory. It's a tarball that contains the headers and compile flags needed
+to build extensions. The extension developer can then include the headers
+and use the compile flags in their build system to build the extension.
+
+Compile flags
+-------------
+
+Convenience files ``cmake.cflags`` (for CMake based projects) and
+``Makefile.cflags`` (for Make-based ones) are provided by the EDK containing
+the needed compile flags. The main one is ``LLEXT_CFLAGS``, which contains the
+flags needed to build an extension. Also provided is a granular set of flags
+that can be used in support of different use cases, such as when building mocks
+for unit tests. The provided flags are:
+
+- ``LLEXT_CFLAGS``: Flags needed to build an extension.
+- ``LLEXT_ALL_INCLUDE_CFLAGS``: Include flags needed to include all headers.
+- ``LLEXT_INCLUDE_CFLAGS``: Include flags for non build generated headers.
+- ``LLEXT_GENERATED_INCLUDE_CFLAGS``: Include flags for build generated headers.
+- ``LLEXT_BASE_CFLAGS``: All other flags needed to build an extension,
+  excluding the include flags.
+- ``LLEXT_GENERATED_IMACROS_CFLAGS``: Include only ``-imacros`` flags for build
+  generated headers.
```

Comment:
This part is a bit difficult to understand as currently written.
After looking at a generated `cmake.cflags`, proposal:
```suggestion
- ``LLEXT_INCLUDE_CFLAGS``: Compiler flags to add non-build-generated headers in include search directories.
- ``LLEXT_GENERATED_INCLUDE_CFLAGS``: Compiler flags to add build-generated headers in include search directories.
- ``LLEXT_ALL_INCLUDE_CFLAGS``: Compiler flags to add all headers in include search directories. (Combination of ``LLEXT_INCLUDE_CFLAGS`` and ``LLEXT_GENERATED_INCLUDE_CFLAGS``)
- ``LLEXT_GENERATED_IMACROS_CFLAGS``: Compiler flags for build-generated headers that must be ``-imacros``-included.
- ``LLEXT_BASE_CFLAGS``: Compiler flags that configure the compiler to emit proper code for the extension. (Note: none of these flags edit the include search directories)
- ``LLEXT_CFLAGS``: All flags required to build an extension. Combination of ``LLEXT_ALL_INCLUDE_CFLAGS``, ``LLEXT_GENERATED_IMACROS_CFLAGS`` and ``LLEXT_BASE_CFLAGS``.
```

Sidenote: if my understanding is correct, the `cmake.cflags` could be simplified to e.g. `set(LLEXT_CFLAGS ${LLEXT_BASE_CFLAGS} ${LLEXT_ALL_INCLUDE_CFLAGS} ${LLEXT_GENERATED_IMACROS_CFLAGS})`? out-of-scope for this PR